### ISSUE TITLE
input event not firing when using multiple cameras.

### ISSUE THREAD

**Issue by samid737** [photonstorm/phaser#3631]
This issue is about: 

A bug in **v3.6.0.**

**UPDATE:** seems to occur in A single scene, so it is probably not related to multiple scenes. please refer to [comment below](https://github.com/photonstorm/phaser/issues/3631#issuecomment-386090747)

There is a parallel scene like in this Phaser example:

[Parralel Scenes](https://labs.phaser.io/view.html?src=src\scenes\parallel%20scenes.js)

Then the following is done in scene A:

* A gameobjectover global input listener is added.
* An extra camera is added.
* The extra camera scrollX is updated.

The expected behaviour for sceneA is according to this Phaser example:

[Multi Cam Sprite](https://labs.phaser.io/edit.html?src=src\input\camera\multi%20cam%20sprite.js)

However, the centre object (rendered to the main camera) does not respond to input the event it seems:

 [example](http://jsfiddle.net/r4rqpuaq/).

The left one, rendered to the extra camera does respond to input events.

Could this be related to #3606 ?

**Comment by samid737** [comment]
f08c610 loops trough each camera during hitTest and returns when over is first populated. Tested with 
A couple of Phaser input examples (inbound today). Any thoughts on this approach? 

Examples:

todo

**Comment by samid737** [comment]
Examples:

* [three cameras](http://jsfiddle.net/7ozoapa0/2/)

Phaser/Input/Camera:
* [Multi Cam Sprite](http://jsfiddle.net/ux2p2ax2/1/)
* [Graphic Contains](http://jsfiddle.net/emb14rtr/2/)

Phaser/Input/GameObject:
* [Multiple Down](http://jsfiddle.net/fur1m177/5/)
* [Top Only Up Down Events](http://jsfiddle.net/v2a6tm8L/1/)

Phaser/Input/Pointer:
* [Multi Camera Cards](http://jsfiddle.net/8rL1atw6/2/)
* [Event List](http://jsfiddle.net/w0hkxsa4/2/)
* [Drag Rectangle(+extra cam)](http://jsfiddle.net/vpfhdq3w/1/)

Phaser/Input/Zones:

* [Drop Zone](http://jsfiddle.net/a6qxqy65/3/)
* [Resize Zone](http://jsfiddle.net/xpLnq7o2/1/)
* [Sprite Drop Zone](http://jsfiddle.net/oc7of5r0/1/)

Phaser/Input/Dragging:

* [Multiple Draggable](http://jsfiddle.net/s52mo8of/2/)
* [Drag With Multiple Scenes](http://jsfiddle.net/yg4or8md/1/)
* [Camera Move And Rotate + extra cam](http://jsfiddle.net/09of1313/3/)

**Comment by photonstorm** [comment]
In the first couple of examples above you're creating extra cameras, but neither of them has been given a position or dimension. So they're effectively "full screen" cameras stacked on-top of each other, which means that only the top-most camera will process the input events (which is what appears to be happening, because if I mouse over the scrolling arrow, the event fires).

During the input tests the pointer will start out by determining which camera it is over (it'll always return the top-most one if you've got a stack of cameras) and only process events for that one camera. It will never iterate down through any other cameras in the list.

So basically, it was working exactly as it was designed to.

However, I can see how it may be useful for the Pointer to iterate the camera list should it not interact with anything in the top-most camera, so I have added this feature in. I'll keep an eye on performance, but I think in most cases (where you only have 1 camera) it should be fine. If it does add undue processing, for what is likely quite an edge-case requirement, I'll swap to using a linked list or similar for the cameras. You can now test this new feature in the master branch.

**Comment by samid737** [comment]
Alright rich, I gradually noticed it too, so it turns out to be more of a feature request indeed.  I'll test the new feature for the [path builder plugin](https://github.com/samid737/phaser3-plugin-pathbuilder) and see if I can remove my [silent and sneaky code override](https://github.com/samid737/phaser3-plugin-pathbuilder/commit/88101146cefe6f5be442add29865864e62ca9d41).
